docs/rfcs/rfc-010-cypher-builder.md

```diff
@@ -108,6 +108,205 @@ actorNode.setAlias("pepe");
 query.print(); // MATCH(pepe:Actor:MyOtherLabel)-[:ACTED_IN]->(m:Movie) ....
 ```
 
+## Patterns
+
+Writing arbitrarily complex patterns (such as `()-[]->()<-[]-()`) can be tricky in a readable and composable API.
+
+**Requirements**
+
+-   Must be composable (i.e. define nodes and relationships and compose into a complex pattern)
+-   Support for node and relationship properties
+-   Parametrized direction
+-   [Variable-length pattern matching](https://neo4j.com/docs/cypher-manual/current/syntax/patterns/#cypher-pattern-varlength)
+-   Path variables
+
+A pattern is a sequence of nodes and relationships, starting at a node.
+
+-   A node contains labels and properties.
+-   A relationship contains a type and properties. Note, relationships no longer contains from and to.
+
+Example patterns
+
+-   `MATCH (m:Movie {title: "The Matrix"})-[:ACTED_IN { role: "neo" }]->(:Person)-[:DIRECTED]-(m2:Movie)`
+-   `MATCH (p:Person)-[*3..5]->(b)`
+-   `(me)-[:KNOWS*1..2]-(remote_friend)`
+-   Assigning to path: `p = (a)-[*3..5]->(b)`
+
+### Proposed solution
+
+**Node**
+
+```typescript
+class Node {
+    related(rel?: Relationship): PartialPattern;
+    withProperties(Record<string, any>): this
+    withoutLabels(): Pattern; // Pattern of only one node without labels
+}
+
+class PartialPattern { // Cannot be compiled
+    to(node?: Node): Pattern;
+    withDirection(direction: "left" | "right" | "undirected");
+}
+
+class Pattern {
+    related(rel?: Relationship): PartialPattern;
+    toPath(): Path; // Extends Variable and can be returned
+}
+```
+
+### Examples
+
+**Long pattern**
+
+```cypher
+MATCH (m:Movie {title: "The Matrix"})-[:ACTED_IN { role: "neo" }]->(:Person)-[:DIRECTED]-(m2:Movie)
+```
+
+```typescript
+const movie = new Cypher.Node({ labels: ["Movie"] });
```

Comment:
mmm, this is tricky, because properties are actually part of the node variable in Cypher, however, it is true that the common use case would be to set them as part of the pattern instead so it may be worth to change how this work in favour of this option :thinking: